modules/integration/tests-platform/tests-clustering/src/test/java/org/wso2/mb/platform/tests/clustering/slot/SlotDeliveringTestCase.java

```diff
@@ -0,0 +1,158 @@
+
+
+		String randomInstanceKey = getRandomMBInstance();
+
+		AutomationContext tempContext = getAutomationContextWithKey(randomInstanceKey);
+
+		AndesJMSConsumerClientConfiguration consumerConfig = new AndesJMSConsumerClientConfiguration(tempContext.getInstance().getHosts().get("default"),
+		                                                                                             Integer.parseInt(tempContext.getInstance().getPorts().get("amqp")),
+		                                                                                             ExchangeType.QUEUE, "clusterSingleQueue1");
+		consumerConfig.setMaximumMessagesToReceived(expectedCount);
+		consumerConfig.setPrintsPerMessageCount(expectedCount / 10L);
+
+
+		randomInstanceKey = getRandomMBInstance();
+		Queue queue = getAndesAdminClientWithKey(randomInstanceKey).getQueueByName("clusterSingleQueue1");
+
+		assertTrue(queue.getQueueName().equalsIgnoreCase("clusterSingleQueue1"), "Queue created in MB node 1 not exist");
+
+		AndesJMSPublisherClientConfiguration publisherConfig = new AndesJMSPublisherClientConfiguration(tempContext.getInstance().getHosts().get("default"),
+		                                                                                                Integer.parseInt(tempContext.getInstance().getPorts().get("amqp")),
+		                                                                                                ExchangeType.QUEUE, "clusterSingleQueue1");
+		publisherConfig.setNumberOfMessagesToSend(sendCount);
+		publisherConfig.setPrintsPerMessageCount(sendCount / 10L);
+
+		AndesClient consumerClient = new AndesClient(consumerConfig, true);
+		consumerClient.startClient();
+
+		AndesClient publisherClient = new AndesClient(publisherConfig, true);
+		publisherClient.startClient();
+
+		AndesClientUtils
+				.waitForMessagesAndShutdown(consumerClient, AndesClientConstants.DEFAULT_RUN_TIME);
+
+		Assert.assertEquals(publisherClient.getSentMessageCount(), sendCount,
+		                    "Message sending failed.");
+		Assert.assertEquals(consumerClient.getReceivedMessageCount(), expectedCount, "Message receiving failed.");
+
+	}
+
+	/**
+	 * Cleanup after running tests
+	 * @throws AndesAdminServiceBrokerManagerAdminException
+	 * @throws RemoteException
+	 */
+	@AfterClass(alwaysRun = true)
+	public void destroy() throws AndesAdminServiceBrokerManagerAdminException, RemoteException {
+
+		String randomInstanceKey = getRandomMBInstance();
+
+		AndesAdminClient tempAndesAdminClient = getAndesAdminClientWithKey(randomInstanceKey);
+
+		if (tempAndesAdminClient.getQueueByName("clusterSingleQueue1") != null) {
+			tempAndesAdminClient.deleteQueue("clusterSingleQueue1");
+		}
+
+		if (tempAndesAdminClient.getQueueByName("clusterSingleQueue2") != null) {
+			tempAndesAdminClient.deleteQueue("clusterSingleQueue2");
+		}
+
+		if (tempAndesAdminClient.getQueueByName("clusterSingleQueue3") != null) {
+			tempAndesAdminClient.deleteQueue("clusterSingleQueue3");
+		}
```

Comment:
If the fix mentioned in "https://github.com/wso2/product-mb/pull/90/files#r28758337" is done, we would only need to delete one queue. I.E the queue created in the test.